File: week-03_05/tp/tp/tp2/ping_pong_1.f90
program ping_pong_1
  use, intrinsic :: iso_fortran_env, only : dp => real64
  USE MPI_F08
  implicit none

  TYPE(MPI_Status)                     :: status
  integer, parameter                   :: nb_values=1000,tag=99
  integer                              :: rank
  real(kind=dp), dimension(nb_values)  :: values
  TYPE(MPI_Datatype)                   :: typedp

  !.TODO: Initialize MPI

    call random_number(values)
    ! TODO: Process 0 send values to process 1

    ! TODO: Process 1 receive values from process 0

    print ('("Me, process 1, received ",i4," values (last = ", &
           & f4.2,") from process 0.")'), nb_values,values(nb_values)

  ! TODO: Finalize MPI

end program ping_pong_1

File: week-03_05/tp/tp_C/tp2/ping_pong_1.c
#include "mpi.h"
#include <stdlib.h>
#include <stdio.h>

int main(int argc, char *argv[]) {
  int rank,iter;
  int nb_values=1000;
  int tag=99;
  double values[nb_values];
  MPI_Status status;

  /* TODO: Init MPI & get rank */

  if (rank == 0) {
    for (iter = 0; iter<nb_values; iter++)
      values[iter] = rand() / (RAND_MAX + 1.);
    /* TODO: Process 0 send values to process 1 */
  } else if(rank == 1) {
    /* TODO: Process 1 receive values from process 0 */

    printf("Me, process 1, received %d values (last = %g)"
           "from process 0.\n", nb_values, values[nb_values-1]);
  }

  /* TODO: Finalize MPI */
  
  return 0;
}
